pep-0605.rst

```diff
@@ -0,0 +1,875 @@
+the rolling releases, that's impossible to answer without consulting a release
+calendar to see when 3.9.0rc1 was branched off from the rolling release series.
+
+By constrast, the equivalent question for rolling beta releases is
+straightforward to answer: "Does Python 3.10.0b2 include all the new features
+released in Python 3.9.0?". Just from formulating the question, the answer is
+clearly "Yes, unless they were provisional features that got removed".
+
+The beta numbering approach also avoids other questions raised by the calendar
+versioning concept, such as how ``sys.version_info``, ``PY_VERSION_HEX``,
+``site-packages`` directory naming, and installed Python binary and extension
+module naming would work.
+
+
+How would users of the rolling beta releases detect API changes?
+----------------------------------------------------------------
+
+When adding new features, core developers would be strongly encouraged to
+support feature detection and graceful fallback to alternative approaches via
+mechanisms that don't rely on either ``sys.version_info`` or runtime code object
+introspection.
+
+In most cases, a simple ``hasattr`` check on the affected module will serve this
+purpose, but when it doesn't, alternative approaches would be considered as part
+of the feature addition. Prior art in this area includes the
+``pickle.HIGHEST_PROTOCOL`` attribute, the ``hashlib.algorithms_available`` set,
+and the various ``os.supports_*`` sets that the ``os`` module already offers for
+platform dependent capability detection.
+
+It would also be possible to add features that need to be explicitly enabled
+via a ``__future__`` import when first included in the rolling beta releases,
+even if that feature flag was subsequently enabled by default before its first
+appearance in an X.Y.0 release candidate.
+
+The rationale behind these approaches is that explicit detection/enabling like
+this would make it straightforward for users of the rolling beta release stream
+to notice when we remove or change provisional features
+(e.g. ``from __future__`` imports break on compile if the feature flag no
+longer exists), or to safely fall back on previous functionality.
+
+The interpreter's rich attribute lookup machinery means we can also choose to
+add warnings for provisional or deprecated imports and attributes that we don't
+have any practical way to add for checks against the value of
+``sys.version_info``.
+
+
+Implications for CPython core development
+-----------------------------------------
+
+The major change for CPython core development is the need to keep the master
+branch more consistently release ready.
+
+While the main requirement for that would be to keep the stable BuildBot fleet
+green, there would also be encouragement to keep the development version of
+the documentation up to date for the benefit of users of the rolling beta
+releases.
+
+On the specific topic of the stable ABI, the "all ones" ``PY_VERSION_HEX``
```

Comment:
Eh, I'd remove this section and just say that if we're not sure the API is stable, don't mark it as stable yet :)

Reply:
Agreed, I think this section might be adding additional complexity to differentiating stable vs questionably stable/unstable when it could be done more simply. Maybe there's a need for it that I'm not seeing though.

Reply:
Done!